### PR TITLE
Fix for issue #94: Escape \u if following 4 characters is not hex digits

### PR DESCRIPTION
**Pull request by janfh**
Suggestion for fixing issue #94

### CODE REVIEW

File: src/serializer.c [hunk 1]
```diff
@@ -51,7 +51,9 @@ static void   jx_EncodeJsonStream (PSTREAM p , PUCHAR in)
 			}
 		}
 		else if  (c  == pJw->backSlash) {
-			if (in[1] != 'u') {  // Dont double escape unicode escape sequence
+			if (in[1] != 'u'   // Dont double escape unicode escape sequence
+				|| !isxdigit(in[2]) || !isxdigit(in[3])
```
Comment (tormodoverlier) [inline src/serializer.c:55]: Could this cause problems if a string ends with for instance `\u` or `\u1`?

Comment (janfh) [reply src/serializer.c:55]: Didn't seem to cause any problems when I tested this. But it did cross my mind as well. If it does, the existing code already did something similar peeking to the next character

Comment (janfh) [reply src/serializer.c:56]: I've added a check on string length before checking the unicode sequence. The length should be at least 6 characters if it contains a unicode sequence.

**Review by tormodoverlier — APPROVED** [review]

**Comment by m1h43l** [comment]
This may also fix #43 .

**Pull request merged**